examples/allocation/zig/testdata/greet.zig

```diff
@@ -0,0 +1,65 @@
+const std = @import("std");
+const allocator = std.heap.page_allocator;
+
+// main is required for Zig to compile to enable WASI.
+pub fn main() anyerror!void {
```

Comment:
Once you use the reactor, I believe we don’t need the main function at all

Reply:
The main is required because the `build.zig` file creates an executable (a runnable binary). If you want to simply call into the Wasm module's functions as a 'library', the `build.zig` file should create a dynamic library instead.
Alternatively, you can overwrite the startup logic by defining your own `_start` function instead like so:
```zig
pub fn export _start() void {
    // do stuff
}
```